pynetdicom/tests/hide_modules.py

```diff
@@ -30,7 +30,7 @@
     import importlib.abc
 
     # py>=3.3 has MetaPathFinder
-    _ModuleHiderBase = getattr(importlib.abc, "MetaPathFinder", importlib.abc.Finder)
+    _ModuleHiderBase = getattr(importlib.abc, "MetaPathFinder")
 except ImportError:
     # py2
     _ModuleHiderBase = object
```

Comment:
While at it, you can remove this try/except, which was only needed for Python 2 compatibility, and directly access the attribute instead of using `getattr`.

It would also make sense to add tests for Python 3.12 (and 3.11, for that matter) in `merge-pytest.yaml` to see this kind of problems in the future.